### ISSUE TITLE
wrong case for naming / namespace

### ISSUE THREAD

**Issue by bsr203** [dotansimha/graphql-code-generator#702]
In the generated code below, you can see the `Message` fragment is referred as `${Message.Document}`, but it should have been `${message.Document}`. 

```ts

export namespace Message {
  export type Fragment = {
    __typename?: 'Message';
    id: string;
    text?: string | null;
    status?: string | null;
    createdAt: Time;
  };
}

export namespace message {
  export const Document = gql`
    fragment message on Message {
      id
      text
      status
      createdAt
    }
  `;
}

export namespace CreateMessage {
  export const Document = gql`
    mutation createMessage($input: CreateMessageInput!) {
      createMessage(input: $input) {
        message {
          ...message
        }
      }
    }

    ${Message.Document}
  `;
  export class Component extends React.Component<
    Partial<ReactApollo.MutationProps<Mutation, Variables>>
  > {
    render() {
      return (
        <ReactApollo.Mutation<Mutation, Variables>
          mutation={Document}
          {...this.props as any}
        />
      );
    }
  }
  export function HOC<TProps = {}>(
    operationOptions?: ReactApollo.OperationOption<TProps, Mutation, Variables>
  ) {
    return ReactApollo.graphql<TProps, Mutation, Variables>(
      Document,
      operationOptions
    );
  }
}
```

my fragment is defined as 

```
export const messageFragment = gql`
  fragment message on Message {
    id
    text
    status
    createdAt
  }
`;

```

I think a better name for namespace would be `messageFragment`. Is it easily configurable?

thanks.

**Comment by ardatan** [comment]
Could you update to these alpha versions?
```
Successfully published:
 - graphql-code-generator@0.13.0-alpha.c5ee0f70
 - graphql-codegen-compiler@0.13.0-alpha.c5ee0f70
 - graphql-codegen-core@0.13.0-alpha.c5ee0f70
 - codegen-templates-scripts@0.13.0-alpha.c5ee0f70
 - graphql-codegen-apollo-angular-template@0.13.0-alpha.c5ee0f70
 - graphql-codegen-introspection-template@0.13.0-alpha.c5ee0f70
 - graphql-codegen-typescript-mongodb-template@0.13.0-alpha.c5ee0f70
 - graphql-codegen-typescript-react-apollo-template@0.13.0-alpha.c5ee0f70
 - graphql-codegen-typescript-template@0.13.0-alpha.c5ee0f70
lerna success publish finished
```

**Comment by bsr203** [comment]
that solved the issue with case sensitivity, but there is two namespace with same name. Hope that is ok.

```
export namespace Message {
	export const Document = gql`
		fragment message on Message {

....

```

```
export namespace Message {
	export type Fragment = {
		__typename?: "Message";
..
```

But, I have bunch of errors (which was present earlier with stable release as well)

```

src/graphql/types.ts:1173:11 - error TS1005: '>' expected.

1173           subscription={Document}
               ~~~~~~~~~~~~

src/graphql/types.ts:1173:23 - error TS1005: ')' expected.

1173           subscription={Document}
                           ~

src/graphql/types.ts:1174:12 - error TS1128: Declaration or statement expected.

1174           {...this['props'] as any}
                ~~~

src/graphql/types.ts:1175:10 - error TS1161: Unterminated regular expression literal.

1175         />


src/graphql/types.ts:1176:7 - error TS1128: Declaration or statement expected.

1176       );
           ~
```

and the offending code is

```
1171 .  return (
1172 .        <ReactApollo.Subscription<Subscription, Variables>
1173 .          subscription={Document}
1174 .          {...this['props'] as any}
 1175 .       />
1176 .      );

```

do I need to change my tsconfig?

**Comment by bsr203** [comment]
Also not sure that change is ok with `"noNamespaces": true` setting. Most of my types are giving duplicate identifier error now as I see same exact fragment def. is duplicated :-(

```
export type MessageFragment = {
	__typename?: "Message";

...

..

export type MessageFragment = {
	__typename?: "Message";
...
```

**Comment by ardatan** [comment]
The file extension must be tsx, not ts if you’re using react template; because it has jsx syntax in it.

**Comment by bsr203** [comment]
sorry. I forgot that. but the file was generated by `graphql-code-generator` though, as I specified the directory. The error went away when I specified a file name with `.tsx` ext. It would be nice if `graphql-code-generator` set the extension if apollo template is used.

**Comment by bsr203** [comment]
so, with `    "noNamespaces": true`, I only have one error `Cannot redeclare block-scoped variable`.

I have a subscription like
```
export const EVENT = gql`
  subscription event {
...
```
and a fragment with same name

```
export const eventFragment = gql`
  fragment event on Event {
    id
    data {
...
``` 

which results in 

```

export const EventDocument = gql`
	subscription event {
		event {
.....




export const EventDocument = gql`
	fragment event on Event {
		id
		data {
....

```

**Comment by ardatan** [comment]
We can think about it but I am not sure how we should solve the problem. But I don’t think using same name for multiple things is a good practice. So I don’t think it is bug, and the only thing we should do is giving a warning or an error to the user like “There are multiple types in your documents” l.
As a workaround you can change their names for now.
@dotansimha what do you think about it?

**Comment by dotansimha** [comment]
@ardatan I agree, it's not a good idea. But I think we can use the duplicates logic to handle this case..

**Comment by ardatan** [comment]
@dotansimha But, in this case types may become unconsistent.

**Comment by dotansimha** [comment]
fixed in 0.13.0. @bsr203
